### PR TITLE
Update event bus to v3.3.1

### PR DESCRIPTION
**Pull request by malinajirka**
This PR updates event bus to v3.3.1. The main change is that the new version doesn't depend on the support library.
https://github.com/greenrobot/EventBus/releases/tag/V3.3.0
https://github.com/greenrobot/EventBus/releases/tag/V3.3.1

To test:
Smoke test the app

## Regression Notes
1. Potential unintended areas of impact
The whole app

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Smoke tested the app - I expect it to either no work anywhere or work everywhere

3. What automated tests I added (or what prevented me from doing so)
The existing tests should cover it

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by develric** [comment]
Hey @malinajirka 👋 , I'm going to test this one and just to be sure I'm on the correct page 😊: 
- testing this one I think is enough to also validate (from WPAndroid side of things) the fluxC related PR [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2210), correct?
- I guess we do not need to change the linked fluxC here because we are assuming gradle will overwrite fluxC dependency to the higher 3.3.0 behind the scene, right?

Just for sharing I'm going to smoke test using the sheet we use in GK as a base and will try to go over the other screens as I can 🙇 .

**Comment by malinajirka** [comment]
> testing this one I think is enough to also validate (from WPAndroid side of things) the fluxC related PR here, correct?

Correct!

> I guess we do not need to change the linked fluxC here because we are assuming gradle will overwrite fluxC dependency to the higher 3.3.0 behind the scene, right?

That's right :)


> Just for sharing I'm going to smoke test using the sheet we use in GK as a base and will try to go over the other screens as I can 🙇 .

Awesome, thank you!

File: WordPress/build.gradle [hunk 1]
```diff
@@ -335,7 +335,7 @@ dependencies {
     implementation 'com.google.android.gms:play-services-auth:18.1.0'
     implementation 'com.android.installreferrer:installreferrer:2.2'
     implementation 'com.github.chrisbanes:PhotoView:2.3.0'
-    implementation 'org.greenrobot:eventbus:3.2.0'
+    implementation 'org.greenrobot:eventbus:3.3.0'
```
Comment (malinajirka) [inline WordPress/build.gradle]: As mentioned [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2210#discussion_r764836506) they released a new version with updated proguard rules - no code changes.
```suggestion
    implementation 'org.greenrobot:eventbus:3.3.1'
```

Comment (malinajirka) [reply WordPress/build.gradle]: cc @develric

**Review by develric — APPROVED** [review]
Smoke tested the app with 3.3.0 (and 3.3.1 :) )...used the sheet from beta testing (not GK as I was wrongly mentioning before :) ) + added a few more smoke testing on notifications, reader, comments etc...everything worked well for me....approving and let you manage the merging sequence when woo also is ready to land 🙇 

PS: I'm not setting approval on the companion fluxc PR just to wait for the woo testing results, but based on what we said [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15664#issuecomment-988704729) feel free to merge that one once woo feedback is positive 🙇

**Pull request merged**